src/mlpack/tests/emfit_test.cpp

```diff
@@ -0,0 +1,223 @@
+/**
+ * @file emfit_test.cpp
+ * @author Kim SangYeon
+ * 
+ * Test for the EMFit class.
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/gmm/em_fit.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::gmm;
+using namespace mlpack::kmeans;
+using namespace mlpack::distribution;
+
+BOOST_AUTO_TEST_SUITE(EMFitTest);
+
+//! Make sure the constructor of EMFit class works correctly.
+BOOST_AUTO_TEST_CASE(EMFitConstructorTest)
+{
+  bool sameCluster = false;
+  bool sameConstraint = false;
+
+  // Build the fitter.
+  EMFit<> fitter;
+
+  // Check the default template parameters of EMFit class.
+  // The EMFit class's default clusterer and constraint for covariance are
+  // KMeans and PositiveDefiniteConstraint classes respectively.
+  if (std::is_same<decltype(fitter.Clusterer()), KMeans<>&>::value)
+    sameCluster = true;
+
+  if (std::is_same<decltype(fitter.Constraint()),
+    PositiveDefiniteConstraint&>::value)
+    sameConstraint = true;
+
+  // Check the default values are correct.
+  // The EMFit class's default max-iteration is 300, and tolerance is 1e-10.
+  BOOST_CHECK_EQUAL(fitter.MaxIterations(), 300);
+  BOOST_CHECK_CLOSE(fitter.Tolerance(), 1e-10, 1e-3);
+  BOOST_CHECK_EQUAL(sameCluster, true);
+  BOOST_CHECK_EQUAL(sameConstraint, true);
+}
+
+/**
+ * Make sure EMFit can estimate parameters reasonably given observations
+ * from the Gaussian distributions.
```

Comment:
Would you say that this is already tested via the `GMMTrainEMMultipleGaussians` test?

Reply:
That makes sense. I thought it would be good to make some unit tests exclusively for `EMFit` class.